core/internal/filetransfer/file_transfer_default.go

```diff
@@ -63,14 +63,12 @@ func (ft *DefaultFileTransfer) Upload(task *Task) error {
 		return err
 	}
 	for _, header := range task.Headers {
-		parts := strings.Split(header, ":")
+		parts := strings.SplitN(header, ":", 2)
```

Comment:
could you add a unit test for this please? (say with a legit-looking aws header, and something else)

Reply:
Sure... or is not done yet...